File: evaluator.ts
import { tokenize } from "./lexer.ts";
import { parse, type ParserState } from "./parser.ts";
import type { EvalError, Result } from "./types.ts";

export function evaluate(input: string): Result<number, EvalError> {
  if (input === "") return { ok: true, value: 0 };

  const state: ParserState = {
    tokens: tokenize(input),
    pos: 0,
    scopes: [new Map()],
    inputLength: input.trimEnd().length,
  };
  return parse(state);
}

File: src/main/kotlin/com/tuff/Evaluator.kt
package com.tuff

fun evaluate(input: String): Int {
    if (input.isEmpty()) return 0
    val tokens = input.trim().split(" ")
    var result = tokens[0].toInt()
    var i = 1
    while (i < tokens.size) {
        val op = tokens[i]
        val value = tokens[i + 1].toInt()
        result = if (op == "+") result + value else result - value
        i += 2
    }
    return result
}
